src/lein_monolith/task/fingerprint.clj

```diff
@@ -77,23 +99,35 @@
               [dir-str]
               ;; Monolith subprojects and profiles don't use absolute paths
               (if (str/starts-with? dir-str (:root project))
-                (jio/file dir-str)
-                (jio/file (:root project) dir-str))))))
+                (io/file dir-str)
+                (io/file (:root project) dir-str))))))
 
 
 (defn- hash-sources
   [project]
   (->> (all-paths project)
        (mapcat list-all-files)
-       (map hash-file)
-       (aggregate-hashes)))
+       (map (fn hash-file
+              [^File file]
+              [(local-path project file)
+               (with-open [in (io/input-stream file)]
+                 (sha1 in))]))
+       (into {})
+       (kv-hash :files)))
 
 
 (defn- hash-dependencies
+  "Hashes a project's dependencies and managed dependencies, as well as that of
+  its profiles and project root."
   [project]
-  (-> (:dependencies project)
-      (pr-str)
-      (->multihash)))
+  (let [hashable-info #(select-keys % [:dependencies :managed-dependencies])]
+    (->> (:profiles project)
+         (map (juxt key (comp hashable-info val)))
+         (into {} (filter (comp seq second)))
+         (merge {::default (hashable-info project)})
+         (puget/render-str (puget/canonical-printer))
+         (str "v1/dependencies\n")
```

Comment:
Seems like you could use `kv-hash` here?

Reply:
I could, but that would require hashing all the individual (fairly small) values first, which felt unnecessary.